lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+            else
+                offsetPos = WorldToLocal(self:GetPos(), angle_zero, ent:GetPos(), angle_zero)
+                offsetYaw = self:GetAngles().y
+            end
+            self:SetData("tracking-offset-pos", offsetPos)
+            self:SetData("tracking-offset-yaw", offsetYaw)
+        end
+
+        if not isTracking then
+            if wasTracking and (not self:GetData("tracking-wasflight")) then
+                self:SetFlight(false)
+            end
+            self:SetData("tracking-wasflight", nil)
+            self:SetData("tracking-offset-pos", nil)
+            self:SetData("tracking-offset-yaw", nil)
+            if IsValid(self.trackingdebugprop) then
+                self.trackingdebugprop:Remove()
+            end
+        end
+
+        if IsValid(ply) then
+            if wasTracking ~= isTracking then
+                TARDIS:StatusMessage(ply, "Controls.Tracking.Status", IsValid(ent))
+            end
+            if isTracking then
+                if wasTrackingEnt ~= ent then
+                    local name = ent.PrintName or (isfunction(ent.Name) and ent:Name()) or ent.Name or ent:GetModel() or ent:GetClass()
+                    if ent.GetCreator then
+                        local creator = ent:GetCreator()
+                        if IsValid(creator) then
+                            name = name .. " (" .. creator:Nick() .. ")"
+                        end
+                    end
+                    TARDIS:Message(ply, "%s %s", "Controls.Tracking.Target", name)
+                else 
+                    TARDIS:Message(ply, "Controls.Tracking.SameTarget")
+                end
+            end
+        end
+        return true
+    end
+
+    function ENT:GetTrackRotation()
+        return self:GetData("tracking-rotation", false)
+    end
+
+    function ENT:SetTrackRotation(on)
+        self:SetData("tracking-rotation", on)
+
+        local ent = self:GetTracking()
+        if not IsValid(ent) then return end
+        local offsetPos = self:GetData("tracking-offset-pos", Vector(0,0,0))
+        local offsetYaw = self:GetData("tracking-offset-yaw", 0)
+        local newOffsetPos, newOffsetYaw
+        if on then
+            local currentTrackingPos = ent:GetPos() + offsetPos
+            newOffsetPos = ent:WorldToLocal(currentTrackingPos)
+            newOffsetYaw = offsetYaw - ent:GetAngles().y
+        else
+            local currentTrackingPos = ent:LocalToWorld(offsetPos)
+            newOffsetPos = WorldToLocal(currentTrackingPos, angle_zero, ent:GetPos(), angle_zero)
+            newOffsetYaw = offsetYaw + ent:GetAngles().y
+        end
+        self:SetData("tracking-offset-pos", newOffsetPos)
+        self:SetData("tracking-offset-yaw", newOffsetYaw)
+    end
+
+    function ENT:ToggleTrackRotation()
+        self:SetTrackRotation(not self:GetTrackRotation())
+    end
+
+    ENT:AddHook("Initialize", "tracking", function(self)
+        self:SetTrackRotation(true)
+    end)
```

Comment:
Are you sure you want this enabled by default?
Considering the fact that most players will not know the switch key, I think it would be best to make it either entity-dependent or false

I'd presume that the most tracked entity would be another TARDIS
Another option is an NPC

NPCs and TARDISes rotating make this one go from side to side rather violently

Reply:
What if we edit the hint for "Tracking enabled" to something like "Tracking enabled, press '<KEY>' to toggle tracking target rotation" or similar?

I don't want to presume what people are going to most likely using this for or try and build up some list of things to turn it on or off for as that will cause inconsistent behaviour for the user, so education on how to toggle it makes sense to me.

Reply:
Okay

Reply:
Added

https://github.com/MattJeanes/TARDIS/pull/969/commits/5e7e592d20c82abb64ad7c17ca179a8bab3b5dad